x/orders/internal/keeper/genesis.go

```diff
@@ -0,0 +1,61 @@
+package keeper
+
+import (
+	"encoding/json"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/dfinance/dnode/x/orders/internal/types"
+)
+
+// InitGenesis inits module genesis state: creates currencies.
+func (k Keeper) InitGenesis(ctx sdk.Context, data json.RawMessage) {
+	k.modulePerms.AutoCheck(types.PermRead)
+
+	state := types.GenesisState{}
+	k.cdc.MustUnmarshalJSON(data, &state)
+	for _, order := range state.Orders {
+		k.set(ctx, order)
+	}
+
+	if err := state.Validate(); err != nil {
+		panic(err)
+	}
+
+	if state.LastOrderId != nil {
+		k.setID(ctx, *state.LastOrderId)
+	}
+}
+
+// ExportGenesis exports module genesis state using current params state.
+func (k Keeper) ExportGenesis(ctx sdk.Context) json.RawMessage {
+	k.modulePerms.AutoCheck(types.PermRead)
+
+	state := types.GenesisState{}
+
+	iterator := k.GetIterator(ctx)
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
```

Comment:
We should wrap this into the func like 'state.Orders = keeper.getOrders()'. That would make code cleaner